sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs

```diff
@@ -33,4 +37,98 @@ public ModelReaderWriterOptions (string format)
     /// Gets the format to read and write the model.
     /// </summary>
     public string Format { get; }
+
+    /// <summary>
+    /// Registers an <see cref="IPersistableModel{T}"/> proxy to be used when reading or writing a model.
+    /// </summary>
+    /// <param name="proxy"> The <see cref="IPersistableModel{T}"/> proxy that will be used to read or write the model. </param>
+    public void AddProxy<T>(IPersistableModel<T> proxy)
+    {
+        _proxies ??= [];
+
+        _proxies.Add(typeof(T), proxy);
+    }
+
+    /// <summary>
+    /// Gets the model that is currently being proxied.
+    /// </summary>
+    public object? ProxiedModel { get; private set; }
```

Comment:
This feels dangerous when we're looking at potential concurrent use.  I know that we generally don't assume model types are thread-safe, but I can see us creating one set of options and passing it to multiple clients.

Reply:
Yeah, this is a good point. I am not even sure the user needs "multiple clients". It's probably enough if the app is a multithreaded app (e.g. ASP.NET) where the same line of code is hit by multiple threads of execution (multiple requests running in parallel).